### PR TITLE
[#111][PR]Added End-to-End test for Camera Based Event Creation

### PR DESCRIPTION
**Pull request by EyalAzimov**
ONLY client-side elements are functioning!

### CODE REVIEW

File: src/components/Calendar/Calendar.js [hunk 1]
```diff
@@ -108,9 +108,9 @@ export default function CalendarView() {
                                bottomLeftRadius='0px' bottomRightRadius='0px' content={calendarMonth}/>
             <StudyRoomChatCard width='92vw' height='6vh' marginTop='2px' topLeftRadius='0px' topRightRadius='0px'
                                bottomLeftRadius='0px' bottomRightRadius='0px' content={<div style={{width:'92vw', display:'flex', justifyContent:"space-between"}}>
-                <BottomDrawer icon={<PrimaryButton2 style={{ margin: 'auto' }} colour={'#057D78'} content={<AddAPhotoIcon/>}/>}
+                <BottomDrawer icon={<PrimaryButton2 data_test={"uplaodPhotoEventButtonCalenderPage"} style={{ margin: 'auto' }} colour={'#057D78'} content={<AddAPhotoIcon/>}/>}
```
Comment (mnashat) [inline src/components/Calendar/Calendar.js:111]: Typo in the word 'upload' in data_test prop. Please fix accordingly.

File: src/components/Calendar/calendar.css [hunk 2]
```diff
@@ -1,6 +1,6 @@
 @import url('https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,500,500i,700,700i,900,900i');
 .events {
-    max-height: 300px;
+    height: 300px;
```
Comment (mnashat) [inline src/components/Calendar/calendar.css:3]: This should be reverted to 'max-height: 300px'. This was a fix for a display issue on the calendar page and should not have any effect on tests.

File: cypress/e2e/SP_29_Course_Study_Hour_Estimator.cy.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
+describe('Test Course Study Hour Estimator [SP-29]', function() {
```
Comment (mnashat) [inline cypress/e2e/SP_29_Course_Study_Hour_Estimator.cy.js:1]: Why are Course Study Hour Estimation tests included in this pull request?

File: cypress/e2e/SP_21_Camera_Based_Event_Creation.cy.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+describe('Test Camera Based Event Creation [SP-21]', function() {
+
+    beforeEach(() => {
+        cy.visit(Cypress.env('baseUrl'));
+        cy.login("testEmail_1","testPassword_1");
+    })
+
+    it('Add Course using Upload Picture Button', function() {
+        cy.get('[data-test="uplaodPhotoEventButtonCalenderPage"]').click();
+        cy.get('#fileSelectButton').invoke('show').selectFile('cypress/photos/Soen_385.png');
```
Comment (mnashat) [inline cypress/e2e/SP_21_Camera_Based_Event_Creation.cy.js:15]: This test is incomplete. It uploads an image, but does not test that the event data from the image is populated in the subsequent addEvent drawer.

**Review by mnashat — CHANGES_REQUESTED** [review]
In addition to the other comments in this review, the image being used for camera based event creation is not a valid image. The images to be used are in PR #109. Please check and update accordingly. Thank you!

File: cypress/e2e/SP_21_Camera_Based_Event_Creation.cy.js (hunk 4 above)
Comment (EyalAzimov) [reply cypress/e2e/SP_21_Camera_Based_Event_Creation.cy.js:15]: True! I couldn't get it working with my image, can we do a walkthrough together with another image?

Comment (mnashat) [reply cypress/e2e/SP_21_Camera_Based_Event_Creation.cy.js:15]: Use the following image to populate all the fields that are supported for this user story:
image

Follow the same steps, but you will noticed that after pressing 'Upload', a new drawer will be presented with the information shown in the image.

**Pull request closed without merging**